File: client/src/typy.ts
export interface Pytanie {
	id: number;
	kategoria?: kategoria;
	pytanie: string;
	odpA: string;
	odpB: string;
	odpC: string;
	odpD: string;
	obrazek: string;
	poprawna: string;
}

export type kategoria = "e12" | "e13" | "e14" | "ee08" | "ee09";

export const wszyskieKategorie: kategoria[] = [
	"e12",
	"e13",
	"e14",
	"ee08",
	"ee09",
];

export type statusType = "sukces" | "blad" | "ladowanie";

File: server/pkg/typy/typy.go
package typy

type Pytanie struct {
	Id        int    `json:"id"`
	Kategoria string `json:"kategoria"`
	Pytanie   string `json:"pytanie"`
	OdpA      string `json:"odpA"`
	OdpB      string `json:"odpB"`
	OdpC      string `json:"odpC"`
	OdpD      string `json:"odpD"`
	Obrazek   string `json:"obrazek"`
	Poprawna  string `json:"poprawna"`
}
